src/backend/distributed/planner/query_colocation_checker.c

```diff
@@ -277,6 +292,14 @@ WrapRteRelationIntoSubquery(RangeTblEntry *rteRelation, List *requiredAttributes
 	RangeTblEntry *newRangeTableEntry = copyObject(rteRelation);
 	subquery->rtable = list_make1(newRangeTableEntry);
 
+#if PG_VERSION_NUM >= PG_VERSION_16
+	if (perminfo)
```

Comment:
So, in this part of the code, we are copying rteRelation, and I passed its perminfo as argument to WrapRteRelationIntoSubquery. However, here we may be dealing with a child relation and it would not have perminfoindex set, meaning that perminfo would be NULL. We only need to do changes if perminfo is not null.